### PR TITLE
Jest testing for to-do-list program

### PR DESCRIPTION
**Pull request by chuanqizhu0820**
* Use jest framework for testing.
* Create a test file ([..].test.js) for a file
* Test the addTask and removeCompletedTaks functions
* Mock a storage object to "imitate" localStorage operations.
* Mock HTML in the tests
* Use describe() method

### CODE REVIEW

File: src/tasks.js [hunk 1]
```diff
@@ -95,9 +89,24 @@ export const editTask = (arr) => {
...
         arr = remainingTask;
         saveAndReload(arr);
-        // localStorage.setItem('tasks', JSON.stringify(new TaskList(arr)));
-        // window.location.reload();
       });
     });
   });
 };
+
+// function add(a, b) {
```
Comment (carlos-ssh) [inline src/tasks.js]: - Kindly remove commented code.

File: src/__tests__/addTask.test.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+/**
+ * @jest-environment jsdom
+ */
+
+// import { add } from 'lodash';
```
Comment (carlos-ssh) [inline src/__tests__/addTask.test.js]: - Kindly remove commented code.

File: src/__tests__/addTask.test.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import addTask from '../__mocks__/addTask.js';
+import localStorageMock from '../__mocks__/localStorage.js';
```
Comment (carlos-ssh) [inline src/__tests__/addTask.test.js:5]: - Fix these errors displayed in console:
Screen Shot 2021-09-29 at 9 54 31

**Review by carlos-ssh — CHANGES_REQUESTED** [review]
Hi @chuanqizhu0820 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__tests__/addTask.test.js (hunk 2 above)
Comment (rloterh) [reply src/__tests__/addTask.test.js]: This issue is well resolved! ✔️

File: src/tasks.js (hunk 1 above)
Comment (rloterh) [reply src/tasks.js]: This issue is well resolved! ✔️

File: src/__tests__/addTask.test.js (hunk 3 above)
Comment (rloterh) [reply src/__tests__/addTask.test.js:5]: This issue is well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Chuanqi, 

Your project is complete! Great job for making the necessary changes recommended by the previous reviewer(s). There is nothing else to say other than... it's time to merge it 🍾 :shipit:.
Congratulations! 🎉🥂

<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 1 more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**